react/javascript/demo-external-api/README.md

```diff
@@ -0,0 +1,113 @@
+# Demo External API
+
+This example demonstrates usage of external API in an extension using javascript. Its uses OAUTH2 to get an access token from Google. It then use that token to create and update a spreadsheet using the Google sheet APIs. The key thing to observe are:
+
+- the OauthProvider which handles interfactions with Googles OAUTH2 identity service. The auth token is stored in the pushRouter state so that it survives a page reload.
+- The SheetsProvider which provides a simple abstraction to calling the Google sheets restful endpoints.
+
+## Prerequisites
+
+This demo requires a project with access to the sheets API and an OAUTH2 client to be setup in the Google console.
+
+1. [Open the Google console](https://console.cloud.google.com/).
+2. Select `IAM and Admin/Create a project`.
+3. Enter the name of the project and click `create` which navigates you to the project dashboard.
+4. On the project dashboard click `select project` for your newly created project. This displays information about your project and tasks you can do.
+5. In the `Getting started` section click `Explore and enable APIs`.
+6. Click `Libraries` and search for `Sheets`. This brings up a link to the `Google Sheets API`. `Select` it.
+7. Click `Enable` which brings you to the Google Sheets API setup.
+8. Click the `Configure Consent Screen` button.
+9. Select the `User Type` and click Create. You maybe restricted as to the type you can select.
+10. Enter `application information` and click `Save and Continue`. The minimum you need to enter is the `App name`, `User support email` and at least one `developer contact email`.
+11. Click `Save and Continue` until you see the `Back to Dashboard` button. Click the `Back to Dashboard` button.
+12. Click the `Credentials` link in the sidebar.
+13. Click the `Create Credentials` button and select `OAUTH client ID`.
+14. Select `Application Type` of `Web Application`.
+15. `Name` your application.
+16. Add the base url of the Looker instance that will host your extension as an `Authorized JavaScript Origin`. Example `https://myinstance.looker.com`.
+17. Add the base url of the Looker instance that will host your extension appended with `/extensions/oauth2_redirect` as an `Authorized redirect URI`. Example `https://myinstance.looker.com//extensions/oauth2_redirect`.
```

Comment:
```suggestion
17. Add the base url of the Looker instance that will host your extension appended with `/extensions/oauth2_redirect` as an `Authorized redirect URI`. Example `https://myinstance.looker.com/extensions/oauth2_redirect`.
```

Reply:
Went through the whole process again. I think I have the definitive version now. I added some comments that hopefully give a hint as to how to fix things if things go wrong